distributed/diagnostics/tests/test_worker_plugin.py

```diff
@@ -53,6 +53,40 @@ async def test_create_with_client(c, s):
     assert worker._my_plugin_status == "teardown"
 
 
+@gen_cluster(client=True, nthreads=[])
+async def test_remove_with_client(c, s):
+    await c.register_worker_plugin(MyPlugin(123), name="foo")
+    await c.register_worker_plugin(MyPlugin(546), name="bar")
+
+    worker = await Worker(s.address, loop=s.loop)
+    # remove the 'foo' plugin
+    await c.unregister_worker_plugin("foo")
+    assert worker._my_plugin_status == "teardown"
+
+    # check that on the scheduler regitered worker plugins we only have 'bar'
+    assert len(s.worker_plugins) == 1
+    assert s.worker_plugins[0]["name"] == "bar"
+
+    # check on the worker plugins that we only have 'bar'
+    assert "foo" not in worker.plugins
+    assert "bar" in worker.plugins
```

Comment:
Instead of checking that `"foo"` isn't in the worker's `dict` of plugins, could we check that `len(worker.plugins) == 1` and that `"bar"` is in `worker.plugins` (similar to what we're doing above with the scheduler)